### PR TITLE
Add device to ToTensor

### LINKED ISSUE #2925: ToTensor Device
**Is your feature request related to a problem? Please describe.**
Currently when converting from other formats that can reside on gpu like cupy to torch tensor, using `ToTensor`, it will create the tensor on the cpu, and then it should be sent to gpu using `ToDevice`. This will create unnecessary between cpu and gpu and reduce the performance. A solution to that is to have `ToTensor` accept device parameter and directly create the tensor on that machine, and in this case on gpu.

### PR DESCRIPTION
**Pull request by drbeh**
Fixes #2925

### Description
Add device to `ToTensor`.

### Status
**Ready**

### Types of changes

- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [ ] Breaking change (fix or new feature that would cause existing functionality to change).
- [ ] New tests added to cover the changes.
- [ ] Integration tests passed locally by running `./runtests.sh -f -u --net --coverage`.
- [ ] Quick tests passed locally by running `./runtests.sh --quick --unittests`.

### CODE REVIEW

File: monai/utils/type_conversion.py [hunk 1]
```diff
@@ -97,26 +97,29 @@ def convert_to_tensor(data, wrap_sequence: bool = False):
 
     """
     if isinstance(data, torch.Tensor):
-        return data.contiguous()
+        if device is None:
```
Comment (Nic-Ma) [inline monai/utils/type_conversion.py:100]: I think maybe no need to check `None`, `to()` can support None:
https://github.com/pytorch/ignite/blob/master/ignite/utils.py#L28
@wyli What do you think?

Thanks.

**Review by wyli — COMMENTED** [review]
thanks, it looks good to me, a minor issue is that it introduces a style warning https://deepsource.io/gh/Project-MONAI/MONAI/run/9269f947-5bda-4b35-80ec-35af55955a22/python/PYL-R1705

File: monai/utils/type_conversion.py [hunk 2]
```diff
@@ -83,7 +83,7 @@ def get_dtype(data: Any):
     return type(data)
 
 
-def convert_to_tensor(data, wrap_sequence: bool = False):
+def convert_to_tensor(data, wrap_sequence: bool = False, device: Optional[torch.device] = None):
```
Comment (Nic-Ma) [inline monai/utils/type_conversion.py:86]: Hi @wyli ,

I am not sure we only support `device` arg or `kwargs` for `torch.to()` or `torch.as_tensor()` here?
What do you think?

Thanks.

File: monai/utils/type_conversion.py (hunk 1 above)
Comment (drbeh) [reply monai/utils/type_conversion.py]: Thanks, fixed.

File: monai/utils/type_conversion.py (hunk 2 above)
Comment (wyli) [reply monai/utils/type_conversion.py:86]: Right, 'to' supports some detailed options, 'as_tensor' doesn't support many, perhaps we can enhance it further in another PR if needed.

Comment (wyli) [reply monai/utils/type_conversion.py:86]: Don't understand this integration error, I'll rerun the test https://github.com/Project-MONAI/MONAI/runs/3569135774

**Review by wyli — APPROVED** [review]
Integration tests work fine

**Pull request merged**